### PR TITLE
Add support for DuckDuckGo

### PR DESCRIPTION
**Pull request by Veritania**
Allows for search queries done via [DuckDuckGo](https://duckduckgo.com/) to replace the `poe` keyword with `site:poewiki.net`, extending the existing google functionality to another search engine.

Tested successfully on Mozilla Firefox 91.0.2

### CODE REVIEW

**Comment by pm5k** [comment]
Hello,

Please can you merge 7ebd71e into your branch since we made some changes to the query parsing. @tonytangio has corrected the parsing to ensure we don't "hijack" non-wiki `poe` queries. Your PR should reflect this, and then, I think I am happy to merge it once corrected. :)

File: src/background/redirects.js
Comment (tonytangio) [inline src/background/redirects.js:41]: (After merging/rebasing from master)

Note how the implementation of both this and `redirectFromGoogle` are identical - you should move these to a common `searchQueryFromRequest` function and remove the duplication entirely.

This will make future search engine support trivial to add.

Comment (tonytangio) [reply src/background/redirects.js:41]: Also, please wait to merge/rebase from master, after #10 has been merged, to minimise any other conflicts. Thanks :)

**Comment by pm5k** [comment]
@Veritania  just getting in touch with you regarding this PR and the progress. If you believe you may not have the time to implement this fully based on the comments above, can you let us know and see if you're comfortable with this being taken over from you on our end so we can pack this change into the future 1.0.7 version? Thanks.

**Comment by Veritania** [comment]
@pm5k Hi, I'll try to give it a go this weekend.

**Comment by Veritania** [comment]
Hi @pm5k 
I've merged in your latest master and updated my branch to be more generic-use and less duplication now, as requested by @tonytangio 

I've also attempted to update the tests, to be easier to add in a new search engine and run the same queries against it. 
I'm open to feedback, and thank you for your time :+1: 

These changes should make it less work to include more search engines in the future.

File: src/background/redirects.spec.js
Comment (tonytangio) [inline src/background/redirects.spec.js:36]: 1. Why not just use a JS object? Map isn't necessary here as you're not dynamically changing these keys+values

2. nit - containers like these maps and objects should be plural (because they hold multiple elements)

3. nit - the keys of this container represent the hosts, but the *values* of this map are URLs that's more than just the host (not sure what I'd call them, perhaps "base"s?)

Here's how I'd approach this (you're welcome to disagree!):

```js
const redirectBaseUrls = {
  google: "https://www.google.com/search?q=site:poewiki.net+",
  duckduckgo: "https://www.duckduckgo.com/?q=site:poewiki.net+"
}
// .get becomes
redirectBaseUrls[searchEngine]
```

**Review by tonytangio — COMMENTED** [review]
👍 Nice work, just have a few small comments and I'm happy to approve after.

File: src/background/redirects.spec.js
Comment (tonytangio) [inline src/background/redirects.spec.js:49]: nit - minor duplication removal/optimization:
```suggestion
                    expected: redirectBaseUrl + "faster+attacks"
```

Under line 44
```js
const redirectBaseUrl = redirectBaseUrls[searchEngine]
```

**Comment by Veritania** [comment]
@tonytangio  Thank you for the feedback. I've implemented the changes you've suggested.
I also added a default to the switch now in 50838c5 . I like that better than having the chance of using a null value in the switch. 

Even if we should never hit the default, you might not agree with it redirecting to ddg instead of google. Privacy vs familiarity in case a user hits the default and suddenly see ddg, a website that might be unknown to them.

**Comment by tonytangio** [comment]
> Even if we should never hit the default, you might not agree with it redirecting to ddg instead of google.

Yeah, I'm leaning towards Google since it's easily the majority choice. Should we ever hit the default case, we don't want most people getting surprised with a different search engine (one they might not even have heard of) and losing trust in the extension.

File: src/background/redirects.js
Comment (tonytangio) [inline src/background/redirects.js:38]: nit: you could just initialise `searchEngine` with the default rather than setting it here, you could even rearrange the fall-throughs like this:

```js
        default:
        case "duckduckgo.com":
        case "www.duckduckgo.com":
            searchEngine = 'https://www.duckduckgo.com/?q=site:poewiki.net+'
 ```

**Review by tonytangio — APPROVED** [review]

File: src/background/redirects.js
Comment (Veritania) [reply src/background/redirects.js]: Fixed in d512a69 , our fallback is google now - I've removed the default all-together as it serves no purpose anymore.

**Pull request merged**
